tests/test_002_CPCandOptimizer.py

```diff
@@ -1211,7 +1208,7 @@ def test_simple_optimizer():
     funcvy = O.optimize(result=O.SO_DXDYVALYFUNC)
     x,y = func0(2100)[0]
     xb, yb, _ = c0.dxdyfromp_f(2100)
-    assert x == xb
+    assert x == xb, f"x={x}, xb={xb}"
     assert y == yb
```

Comment:
Consider adding an assertion-message here too.
Also, consider making both of these messages more verbose (which will also make it easier to find each them in the code).